docs/tls-proxy-forwarder.md

```diff
@@ -0,0 +1,73 @@
+## Enable TLS communication between proxy and agent-protocol-forwarder
+
+To enable TLS, you'll need the following:
+
+- Client certificate and key: This is for the proxy to use.
+  The files must be named as `client.crt` and `client.key`.
+- Server certificate and key: This is for the agent-protocol-forwarder running inside the Pod VM.
+  The `SubjectAlternateName (SAN)` of the certificate should have the name `podvm_server`.
+  The files must be named as `tls.crt` and `tls.key`.
+- CA certificate (Optional): This is required if you are using self-signed certificates.
+  The file must be named as `ca.crt`
+
+The file name restrictions for certificates is due to the current podvm image generation and deployment.
+This will be removed in a future release.
+
+Please note that tls enablement described in this document is specific to the control
+plane communication between `cloud-api-adaptor` that is running in the K8s worker
+node and the `agent-protocol-forwarder` that is running in the Pod VM. More
+specifically the communication between the proxy, that is run via
+`cloud-api-adaptor` and the `agent-protocol-forwarder` that is running inside the
+Pod VM will be using TLS.
+
+### Add the server certificate to the Pod VM image
+
+- Copy the server certificate, key and (optionally) the CA certificate to `~/cloud-api-adaptor/podvm/files/etc/certificates`.
+You will need CA certificate if using self-signed certificates.
```

Comment:
Do we need `.gitignore` for these files?

Reply:
Yeah.. better to ignore the certificate files inside the directory